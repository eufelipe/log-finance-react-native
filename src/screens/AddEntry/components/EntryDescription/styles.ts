import styled from 'styled-components/native';

import Colors from 'styles/colors';
import {Text} from 'styles/layout';

export const Label = styled(Text)`
  color: ${({theme}) => theme.colors.white}50;
  font-size: ${({theme}) => theme.fontSizes.small};
`;

export const Input = styled.TextInput.attrs({
  numberOfLines: 1,
  placeholderTextColor: Colors.grayLight,
  blurOnSubmit: true,
})`
  font-family: ${({theme}) => theme.fontFamily.regular};
  font-size: ${({theme}) => theme.fontSizes.small};
  color: ${({theme}) => theme.colors.grayLight};
  font-weight: ${({theme}) => theme.fontWeight.bold};
`;
