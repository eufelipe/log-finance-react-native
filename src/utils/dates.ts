import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
