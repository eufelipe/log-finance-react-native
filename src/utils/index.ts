export {default as number} from './number';
export {default as strings} from './strings';
