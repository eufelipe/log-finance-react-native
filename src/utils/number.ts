export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
