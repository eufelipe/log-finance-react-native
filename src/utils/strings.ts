export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
