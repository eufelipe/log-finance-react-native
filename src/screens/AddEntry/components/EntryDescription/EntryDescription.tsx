import React from 'react';
import {useTranslation} from 'react-i18next';

import Row from '../Row';

import {Label, Input} from './styles';
import {useEntry} from 'hooks/useEntry';

const EntryDescription = (): JSX.Element => {
  const {description, setDescription} = useEntry();
  const {t} = useTranslation('add');

  return (
    <Row>
      <Label>{t('description')}</Label>
      <Input
        onChangeText={setDescription}
        value={description}
        placeholder={t('description-placeholder')}
      />
    </Row>
  );
};

export default EntryDescription;
