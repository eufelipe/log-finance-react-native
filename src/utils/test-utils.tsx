/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
