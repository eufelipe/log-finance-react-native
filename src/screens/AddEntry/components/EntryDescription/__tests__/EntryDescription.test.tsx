/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import EntryDescription from '../EntryDescription';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: (key: any) => key}),
}));

describe('EntryDescription', () => {
  it('Test match snapshot EntryDescription', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<EntryDescription {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
