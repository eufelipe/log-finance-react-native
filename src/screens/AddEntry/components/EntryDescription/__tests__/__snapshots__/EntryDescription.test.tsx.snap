// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryDescription Test match snapshot EntryDescription 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E1E3E510",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 42.857142857142854,
        "marginRight": 42.857142857142854,
        "marginTop": 21.428571428571427,
        "paddingBottom": 42.857142857142854,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff50",
            "fontFamily": "Poppins-Regular",
            "fontSize": 28,
            "fontWeight": "400",
          },
        ]
      }
    >
      description
    </Text>
    <TextInput
      blurOnSubmit={true}
      numberOfLines={1}
      placeholder="description-placeholder"
      placeholderTextColor="#E1E3E5"
      style={
        Array [
          Object {
            "color": "#E1E3E5",
            "fontFamily": "Poppins-Regular",
            "fontSize": 28,
            "fontWeight": "700",
          },
        ]
      }
    />
  </View>
</View>
`;
